### DISCUSSION TITLE
Rotor free-wheeling causes torque moments

### DISCUSSION THREAD

**Discussion by ermarch** [JSBSim-Team/jsbsim#585] [Q&A]
Hi,

When analyzing @dpculp Calidus auto gyro model I noticed the model started spinning at the moment the rotor starts spinning (while it's not being driven by a power source).

Ultimately this let me to the conclusion that JSBSim needs a patch like the following for free-wheeling operations:
```diff
diff --git a/src/FDM/JSBSim/models/propulsion/FGRotor.cpp b/src/FDM/JSBSim/models/propulsion/FGRotor.cpp
index ab064e59f..f6bd5ce60 100644
--- a/src/FDM/JSBSim/models/propulsion/FGRotor.cpp
+++ b/src/FDM/JSBSim/models/propulsion/FGRotor.cpp
@@ -534,6 +534,10 @@ void FGRotor::calc_torque(double theta_0)
            (1.0+4.5*sqr(mu))/8.0
                      - (Thrust*lambda + H_drag*mu)*Radius;
 
+  if (!Transmission->GetFreeWheelTransmission()) {
+    Torque = -PolarMoment;
+  }
+
   return;
 }
```

Now, I'm by no means an export on anything helicopter related but at least this gives me a steady motion even with a rotating rotor.

**Comment by bcoconni** [comment]
> I noticed the model started spinning at the moment the rotor starts spinning (while it's not being driven by a power source).

Any idea about what made the rotor start spinning in the first place ? Since you mentioned there was no power input, I'd guess that it would be windmilling or something like this ?

**Reply by ermarch** [reply]
Yes it is auto rotation/windmilling.

It's a simple test, start a rotor aircraft at altitude and watch what happens:

Without this patch the rotor starts rotating and the aircraft starts yawing to the right in an uncontrollable way.
With this patch the rotor starts rotating and the aircraft keeps straight while descending.

I was reading up on the freewheeling device and it's also called a torque inverter. Which made me conclude something like the patch is required.

**Reply by ermarch** [reply]
Never mind, it turned out to be a configuration mistake: the flapping moment was 100x to high.

**Comment by ermarch** [comment]
In addition to that, Torque can be disabled by setting the sense parameter to zero:

The 'sense' parameter from the thruster is interpreted as follows, sense=1 means
    counter clockwise rotation of the main rotor, as viewed from above. This is as a far
    as I know more popular than clockwise rotation, which is defined by setting sense to
    -1. Concerning coaxial designs - by setting 'sense' to zero, a Kamov-style rotor is
    modeled (i.e. the rotor produces no torque).

**Reply by bcoconni** [reply]
Good to know. Thanks.
